java/vector/src/main/java/org/apache/arrow/vector/stream/MessageSerializer.java

```diff
@@ -120,6 +128,12 @@ public static ArrowBlock serialize(WriteChannel out, ArrowRecordBatch batch)
 
     long start = out.getCurrentPosition();
     int bodyLength = batch.computeBodyLength();
+    // make sure that the body ends on an 8 byte boundary
+    // each buffer in the body gets started on an 8 byte boundary in ArrowRecordBatch,
+    // but we need to handle the final end alignment here
+    if (bodyLength % 8 != 0) {
+      bodyLength += 8 - (bodyLength % 8);
+    }
```

Comment:
Any reason not to drop this in favor of an assertion that `bodyLength` is a multiple of 8? Anything else is a bug IMHO; `batch.computeBodyLength()` should not return a value that is not padded to 8 byte multiple. I'm going to add some DCHECK (debug assertions) to the C++ implementation in a little bit